### PR TITLE
Speed up `update_weights_from_tensor`

### PR DESCRIPTION
**Pull request by fzyzcjy**
## Motivation

Speed up by avoiding tensor serialization and deserialization. Since it was only used for tests before and never released, the old one was OK, and the new one with API breaking changes seems OK.

## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by fzyzcjy** [comment]
CI looks green now

File: test/srt/test_update_weights_from_tensor.py [hunk 1]
```diff
@@ -1,32 +1,40 @@
+import time
 import unittest
 
 import torch
 
 import sglang as sgl
-from sglang.test.test_utils import DEFAULT_SMALL_MODEL_NAME_FOR_TEST
+from sglang.test.test_utils import DEFAULT_MODEL_NAME_FOR_TEST
```
Comment (merrymercy) [inline test/srt/test_update_weights_from_tensor.py]: can we use the small model for faster test?

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: test/srt/test_update_weights_from_tensor.py (hunk 1 above)
Comment (fzyzcjy) [reply test/srt/test_update_weights_from_tensor.py]: Sure, I originally used the bigger one to test speedup and forget to change back

Comment (fzyzcjy) [reply test/srt/test_update_weights_from_tensor.py]: Done (waiting for CI)

**Comment by fzyzcjy** [comment]
Hmm, quick question firstly: is it possible this is flakyness? Commit 5867d7a8f062e433ea09e41cdb18f4375069ae72 fails accuracy-test-2, while 8f8c9c1c758d560621b83e6cc537cac95d3890f6 (same logic) fails accuracy-test-1. And the failures does not seem to be super related to the current PR at first glance.

**Comment by merrymercy** [comment]
@fzyzcjy Yes, some these are flaky.

**Comment by fzyzcjy** [comment]
I see

File: python/sglang/srt/managers/tp_worker.py [hunk 2]
```diff
@@ -197,7 +197,7 @@ def update_weights_from_distributed(
 
     def update_weights_from_tensor(self, recv_req: UpdateWeightsFromTensorReqInput):
         success, message = self.model_runner.update_weights_from_tensor(
-            recv_req.name, recv_req.tensor
+            MultiprocessingSerializer.deserialize(recv_req.serialized_named_tensors)
```
Comment (chaokunyang) [inline python/sglang/srt/managers/tp_worker.py:200]: @fzyzcjy Will this work for inter-node cases? For example, if tp_size is greater than 8

Comment (fzyzcjy) [reply python/sglang/srt/managers/tp_worker.py:200]: Yes as long as your tensors are from correct devices
